Clarify that `base_url` in client `well_known` may or may not include trailing slash (#3562)
matrix-org/matrix-doc: content/client-server-api/_index.md

        1.  If the content cannot be parsed, then `FAIL_PROMPT`.
    4.  Extract the `base_url` value from the `m.homeserver` property.
        This value is to be used as the base URL of the homeserver.
        1. If this value is not provided, then `FAIL_PROMPT`.

>         2. It is important to note that this value may or may not include
>            a trailing `/`. Consumers should be prepared to handle both 
>            cases.

Reviewer: We've found that "may" introduces more confusion than not in the formal spec:
```suggestion
        2. It is important to note that this value might not include
           a trailing `/`. Consumers should be prepared to handle both 
           cases.
```

        1.  If the content cannot be parsed, then `FAIL_PROMPT`.
    4.  Extract the `base_url` value from the `m.homeserver` property.
        This value is to be used as the base URL of the homeserver.
        1. If this value is not provided, then `FAIL_PROMPT`.
        2. It is important to note that this value might include a trailing

>            `/`. Consumers should be prepared to handle both cases.

Reviewer 2: I think we need to move this a bit further down - it doesn't seem to make much sense to talk about trailing slashes before validating it as a URL?
